runtime/protocol/aws-json-protocols/common/src/aws/smithy/kotlin/runtime/awsprotocol/json/RestJsonErrorDeserializer.kt

```diff
@@ -46,15 +46,17 @@ public object RestJsonErrorDeserializer {
     }
 
     public fun deserialize(headers: Headers, payload: ByteArray?): ErrorDetails {
-        var code: String? = headers[X_AMZN_ERROR_TYPE_HEADER_NAME]
-        var message: String? = if (headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME] == null) headers[X_AMZN_EVENT_ERROR_MESSAGE_HEADER_NAME] else headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME]
+        var message = headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME] ?: headers[X_AMZN_ERROR_MESSAGE_HEADER_NAME]
+        val headerCode: String? = headers[X_AMZN_ERROR_TYPE_HEADER_NAME]
+        var bodyCode: String? = null
+        var bodyType: String? = null
```

Comment:
**style**: You can use `lateinit` to avoid setting `null` here

Reply:
I think `lateinit` would be unsafe here since the values are not guaranteed to be set. The payload may be null, the descriptors may not be found, etc.